submariner/globalnet-ovn.md

```diff
@@ -0,0 +1,131 @@
+   chains in the `filter` table.
+   * On the active Gateway node of the cluster, these chains are programmed with necessary IPtable rules to allow the
+   inter-cluster traffic.
+   * When a remote endpoint notification is received by the handler, it programs routes in `table 150` on all the
+   worker nodes with default next hop present on the `ovn-k8s-mp0` interface.
+   Basically, this is done to support pods with hostNetworking to connect to a remote service.
+   The idea here is to send the traffic via the CNI network and not via the default interface on the host.
+   * Similarly, on the Gateway node, it programs
+     * routes in `table 149` to redirect the incoming traffic from the remote cluster via the `submariner_router` and
+     * IPtable rules in `SUBMARINER-POSTROUTING` chain of `nat` table not to perform any NAT operation on this traffic.
+
+![SubmarinerOVNTopology](./images/ovn-submariner-architecture.svg)
+
+So, in the Submariner OVN architecture the inter-cluster traffic touches the host before leaving the Gateway node via
+the tunnel. Because of this, we can leverage the current Globalnet design which programs the iptable rules on the
+Gateway node even for OVN-Kubernetes deployments.
+
+Also, OVN-Kubernetes controller itself uses IPtables for the following use-cases.
+
+* For supporting NodePort based services.
+* For supporting regular clusterIP services configured with `external-ips`, the OVN controller programs IPTable rule
+in `PREROUTING` chain of `nat` table to perform DNAT operation from `external-ip` to the corresponding `cluster-ip`
+associated with the service.
+
+Given that both Submariner and OVN-Kubernetes already use iptables for a couple of use-cases, this option is the
+most suitable and safe option among the ones we evaluated for supporting OVN-Kubernetes deployments.
+
+### Changes
+
+The following fixes are necessary in Submariner to support Globalnet on OVN-Kubernetes clusters.
+
+* [Missing rule in SUBMARINER-POSTROUTING chain](https://github.com/submariner-io/submariner/issues/1981)
+* [Missing default route in table 150 on all the worker nodes](https://github.com/submariner-io/submariner/issues/1982)
+* [Redirect the ingress traffic to a cluster via the submariner_router](https://github.com/submariner-io/submariner/issues/1983)
+* [Fix logical routes programmed in submariner_router](https://github.com/submariner-io/submariner/issues/1984)
+
+### Pros
+
+* Since the same Globalnet code will be used even for OVN-Kubernetes, it will be easy to maintain.
+* Any enhancements/fixes that go into Globalnet controller will benefit all the CNIs.
+* It will take less effort to support Globalnet for OVN-Kubernetes deployments.
+
+### Cons
+
+* Nothing in particular.
+
+### Backward Compatibility
+
+Now to support this, we require a couple of changes in Network Plugin Syncer and OVN route-agent plugin code.
```

Comment:
Do you mean the changes mentioned above?

Reply:
Yes, that's correct Yossi.